pkg/server/grpc/server.go

```diff
@@ -88,6 +88,59 @@ func (s *Server) Run() error {
 }
 
 // New creates a new instance of `IstioCAServiceServer`.
-func New(ca ca.CertificateAuthority, port int) *Server {
-	return &Server{ca, port}
+func New(ca ca.CertificateAuthority, hostname string, port int) *Server {
+	return &Server{
+		ca:       ca,
+		hostname: hostname,
+		port:     port,
+	}
+}
+
+func (s *Server) createTLSServerOption() grpc.ServerOption {
+	config := &tls.Config{
+		GetCertificate: func(*tls.ClientHelloInfo) (*tls.Certificate, error) {
+			if s.certificate == nil || !isValid(s.certificate) {
+				// apply new certificate if there isn't one yet, or the one has become invalid
+				newCert, err := s.applyServerCertificate()
+				if err != nil {
+					return nil, err
+				}
+				s.certificate = newCert
+			}
+			return s.certificate, nil
+		},
+	}
+	return grpc.Creds(credentials.NewTLS(config))
+}
+
+func (s *Server) applyServerCertificate() (*tls.Certificate, error) {
+	opts := ca.CertOptions{
+		Host:       s.hostname,
+		RSAKeySize: 1024,
```

Comment:
Thought 1024-bit RSA is not considered secure enough [(ref)](https://knowledge.rapidssl.com/support/ssl-certificate-support/index?page=content&id=INFO1684&actp=search&viewlocale=en_US&searchid=1442967219430), use 2048?